src/os_crypto/md5_sha1_sha256/md5_sha1_sha256_op.c

```diff
@@ -61,8 +62,36 @@ int OS_MD5_SHA1_SHA256_File(const char *fname, const char *prefilter_cmd, os_md5
     SHA256_Init(&sha256_ctx);
 
     /* Update for each one */
+#ifdef WIN32
     while ((n = fread(buf, 1, 2048, fp)) > 0) {
+#else
+    while ((n = w_fread_timeout(buf, 1, 2048, fp, 5)) > 0) {
```

Comment:
Buffer limit `2048` should not be hardcoded.
Please, use a local variable at the beginning `fread_buffer`